### PR TITLE
Making the mocking mechanism easier to switch on and off

### PR DESCRIPTION
**Pull request by rvisuvasam**
Alright, here we go with some controversial changes:

1) I don't like the `SHOULD_MOCK` flag's location, and I would love for a suggestion to improve that
2) I've started to remove the fixed values from the specs themselves (only `start_session` so far)

### CODE REVIEW

**Comment by ufchrisg** [comment]
Why have a flag? Mocking should always be happening in unit tests for external resources.

**Comment by rvisuvasam** [comment]
@ufchrisg -- I was thinking about the case that a future developer would like to actually interact with the ThingSpace API, and bypass mocking (something I recently experienced).  I basically want to change one flag, and have all my tests run against the actual service instead of mocked responses.

File: spec/support/helpers.rb [hunk 1]
```diff
@@ -1,21 +1,36 @@
...
-  # NOTE: Using these values as a shared set of default values
-  def base_mock_status
-    "success"
+  def set_mock_status(new_mock_status)
+    request_options[:mock_status] = new_mock_status if SHOULD_MOCK
+  end
+
+  def base_request_options
+    {
+      api_key: "<PLACE API KEY HERE>",
```
Comment (rvisuvasam) [inline spec/support/helpers.rb]: Someone smarter than me make this safer than this, please?!  Would be way too easy to commit real creds if testing.  Maybe using environment variables is the way to got (using `dotenv` or something).

**Comment by rvisuvasam** [comment]
Before everyone yells at me for making a massive PR -- just know that I had my reasons (mainly lack of sleep, and a bag of potato chips).

**Pull request closed without merging**